umake/frameworks/__init__.py

```diff
@@ -340,11 +340,59 @@ def load_module(module_abs_name, main_category):
     if current_category not in BaseCategory.categories.values():
         return
     for framework_name, FrameworkClass in inspect.getmembers(module, _is_frameworkclass):
-        if FrameworkClass(current_category) is not None:
+        if FrameworkClass(category=current_category, force_loading=force_loading) is not None:
             logger.debug("Attach framework {} to {}".format(framework_name, current_category.name))
 
 
-def load_frameworks():
+def list_frameworks():
+    """ List all frameworks.
+
+        Return a dictionnary. Format of it is:
+        [
+            {
+                'category_name':
+                'category_description':
+                'frameworks':
+                    [
+                        {
+                            'framework_name':
+                            'framework_description':
+                            'install_path': None or Path string
+                            'is_installable': True or False
+                            'is_category_default': True or False
+                            'only_for_removal': True or False
+                        },
+                    ]
+            },
+        ]
+    """
+    categories_dict = list()
+    for category in BaseCategory.categories.values():
+        frameworks_dict = list()
+        for framework in category.frameworks.values():
+            new_fram = {
+                "framework_name": framework.prog_name,
+                "framework_description": framework.description,
+                "install_path": framework.install_path if framework.is_installed else None,
```

Comment:
what's the default if framework is not installed? If we have framework.install_path returning "" or None, we can just use this and delete the if condition.